doc/group_instances.md

```diff
@@ -0,0 +1,269 @@
+### Group Instance & Group Instance IDs
+
+---
+
+It is recommended that you familiarise yourself with [*repeating routing rules*](repeating_routing_rules.md) before proceeding with this.
+
+Group Instance / Instance IDs are used to identify the instance of an answer acquired from a repeating group.
+
+#### Group Instance (Index)
+An answer can have multiple instances, the *group_instance* tracks this instance and is part of the [*Location*](../app/questionnaire/location.py) object which consists of *group_id*, *group_instance*, and the *block_id*.
+
+Currently, `group_instance` is always used as it is part of the Location. Any non-repeating answers will have a `group_instance` of `0` by default.
+
+##### Group Instance - Repeats
+
+On every request to `/questionnaire/<eq_id>/<form_type>/<collection_id>/`, we generate the full routing [*Path*](../app/questionnaire/path_finder.py) which holds a list of *Location* objects.
+In short, every block in every group is added to the path. Any block or group with skip conditions or routing rules which evaluates *True* is not added to the path.
```

Comment:
This doesn't quite make sense to me. I think something like this might be better?

`Starting from the first block in the questionnaire, the routing path is created by following routing rules and evaluating skip conditions until the end of the survey is reached.`